api/all/src/main/java/io/opentelemetry/api/logs/LogRecordBuilder.java

```diff
@@ -98,9 +100,25 @@ default LogRecordBuilder setAllAttributes(Attributes attributes) {
     return this;
   }
 
+  /** TODO. */
+  @SuppressWarnings("unchecked")
+  default LogRecordBuilder setAllAttributes(ExtendedAttributes attributes) {
+    if (attributes == null || attributes.isEmpty()) {
+      return this;
+    }
+    attributes.forEach(
+        (attributeKey, value) -> setAttribute((ExtendedAttributeKey<Object>) attributeKey, value));
+    return this;
+  }
+
   /** Sets an attribute. */
   <T> LogRecordBuilder setAttribute(AttributeKey<T> key, T value);
 
+  /** TODO. */
+  default <T> LogRecordBuilder setAttribute(ExtendedAttributeKey<T> key, T value) {
```

Comment:
I suspect supporting setters for both kinds of attributes mitigates the primary concern here: https://github.com/open-telemetry/opentelemetry-specification/issues/4201

it could also be a bit confusing when iterating over the extended attribute and testing if something is equal to one of our semconv constants, but that's only an issue when implementing SDK processors, so definitely not as important